### PR TITLE
enh-test: bare min travis environment

### PR DESCRIPTION
**Pull request by sem-geologist**
This addresses: #1146

This adds additional travis linux environment
to the matrix which:
* does not install conda
* does not install cython, lxml, and anything which is not in setup dependency.
* uses available system `python3` and mimics most close the
  fresh installation using system available `pip3`.

Why it is needed?
on one side:
* hyperspy is available through pip
* documentation declares that it is one of the way to install it using pip.

on the other side:
* it does not work so perfectly if tried on fresh non-conda system
* some of the requirements are declared to be optional, but it's optionality is ignored in some
of tests and parts of hyperspy.

TASKS:

- [x] modify the `.travis.yml` to include one more matrix element, which tries to install hyperspy without conda.
- [x] fix the build/test/execution failures resulting from treating some libraries as (or not treating as) optional (contribution of fixes are more than welcome), either code should be fixed or, either the libraries should be included to dependencies list!

Things to fix:
- [x] `fast_svd` import error
- [ ] is `skimage` optional or not? lookign at signal class it is not, however it is missing in the dependency of setup.

### CODE REVIEW

**Comment by sem-geologist** [comment]
my first feedback to the first try:
it is failing!!!
* it is replicating the bug #1338 which is mystically resolved (so that is why it was closed) after installing/uninstaling sklearn/skimage, and closing and launching notebook again. This stinks, the failing stinking part should be identified and fixed, think about potential user, if he/she can't get it working from the start, probably after minute he/she is going to uninstall it... (and I run into it on fresh linux install #1338), btw the culprit is `fast_svd`.
* ~~dear maintainer of `digital_micrograph` plugin, your test are over ^ 2 verbose; leave debugging info out from travis, as (1) it is irrelevant, because it is even not the culprit i.e. of this failure – in other similar failure case it would hide the real place of failure by overflowing logs (2) debugging should not be done on the travis. In our case Travis built-in log viewer complains that it is too long, but even the downloaded log version is truncated at the end as it overflows 5MB. It also freezes travis tests at that 5MB overflowing event...~~ update: it looks hyperspy.api should set logging level, but this problem apperas as it does not manage to be loaded due to above...

I think this will need a lot of work, but mess should be cleaned.
I propose that things should be fixed in separate PR, in the meanwhile this PR will be updating to upstream

~~update:~~
~~it is possible that there are some other over-verbose plug-ins, just We cant see it as digital_micrograph is tested first and log is flooded by it. So generally:~~
~~* note should be added to doc for developers that "if plugin produces debug information, tests should be set to log them not".~~
~~* fix it in existing plugins.~~

**Comment by sem-geologist** [comment]
should coveralls be also called after minimal environment testing finished?
I wish it would cumulate the coverage, then coverage would rise up, as some of stuff (especially those test skips in full tests, missing libraries conditions) were ignored previously.

**Comment by sem-geologist** [comment]
@francisco-dlp , is skimage optional library? From failing tests it does not look like that. If it is intended to be optional, then something like lxml check inbcf should be added.

**Comment by francisco-dlp** [comment]
Very nice initiative. I think that we should aim at decreasing the strong dependencies of HyperSpy to the bare minimum. So, you're right, skimage is currently a dependency but I think that it'll be sensible to  turn it into a recommend library.

**Comment by sem-geologist** [comment]
skimage is used at least in two places: tiff io_plugin and _signal.complex;
first one treats it as optional, second one as obviously present.
so maybe it should be treated as dependency?

I think I should add scikit-image to setup.py install_req, shaving to the min should be other PR.
This PR is to setup workflow and tool which shows if install_req in setup.py is consistent with reality.

and if talking about shaving to min, I would start from completely other stuff like traistui and other thingies which have nothing to do with algorithms, but is used for (G)UI's - things where scripts does not need any of them.

**Comment by francisco-dlp** [comment]
I fully agree: HyperSpy shouldn't need any plotting or gui package to run. Actually, fully getting rid of traitsui may be desirable as it is by far the package causing more installation troubles.

**Comment by sem-geologist** [comment]
Nice!!! coverall cumulates coverage so now even loops/conditions, which is for skipping tests or imports, gets covered. +0.15% is not much but it is nice side effect :)

**Comment by francisco-dlp** [comment]
Nice indeed. Is it ready for review?

By the way, why not submitting it to RELEASE_next_patch? You are even fixing a couple of bugs (as it always happens when adding unittests ;) )

**Comment by sem-geologist** [comment]
it is ready for review, I think I mixed up the branches, now I am scratching my head... how to correct it?

**Comment by francisco-dlp** [comment]
You could use [rebase or cherrypick](http://stackoverflow.com/questions/1994463/how-to-cherry-pick-a-range-of-commits-and-merge-into-another-branch) and, after pushing the changes, push the edit button at the right of the title and change the base to RELEASE_next_patch.

**Comment by sem-geologist** [comment]
ups... I need to rebase first..

File: .travis.yml [hunk 1]
```diff
@@ -3,42 +3,67 @@ language: python
...
-  - python setup.py install
+  - if [[ $MINIMAL_ENV == 'False' ]] ; then
+      DEPS="nose numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel";
+      conda create -n testenv --yes $DEPS;
+      source activate testenv;
+      conda install pip;
+      pip install coverage coveralls traits traitsui requests tqdm;
+      python -c "import sys; print(sys.executable)";
+      python setup.py install;
+    else pip3 install .;
```
Comment (vidartf) [inline .travis.yml]: Why not move the coverage and coveralls installs for minimal case up here?

Comment (sem-geologist) [reply .travis.yml]: Imho `pip install coverage coveralls...` should be  the same moved to `script` where it is needed. This way time is optimized, so if installation fails, no time is spent then for installing stuff it is not going to be used. I understand the urge to have it organized into travis present `before_install`, `install`, `script`... sections, but build/testing is all ready taking too long time, You can get nap while waiting for travis to finish... And sometimes you need to wait quite long only to get it to fail at installation/build step before even testing starts.

**Comment by sem-geologist** [comment]
I tried todo rebase, but it does not work :/ I give up....

**Comment by vidartf** [comment]
Run this command (assuming you have added `hyperspy` as the remote for the main repo):
```
git rebase --onto hyperspy/RELEASE_next_patch hyperspy/RELEASE_next_minor patch-14
```

It rebased without any conflicts for me.

**Comment by sem-geologist** [comment]
yes it do same as my trial of doing it, it still gets whole list of commits between next_minor and next_patch branches...

**Comment by sem-geologist** [comment]
Maybe i simply close this and do proper pull request... I have no idea why git is for me like thread for a cat...

**Comment by vidartf** [comment]
Do you do `git push -f` afterwards?

**Comment by sem-geologist** [comment]
Maybe i simply close this and do proper pull request... I have no idea why git is for me like thread for a cat...

**Comment by sem-geologist** [comment]
I did `git pull` and `git push`

**Comment by vidartf** [comment]
Do these steps instead:
```
git rebase --onto hyperspy/RELEASE_next_patch hyperspy/RELEASE_next_minor patch-14
git push -f
```

No pull after rebase, and the `-f` option is important for the push.

**Comment by sem-geologist** [comment]
I get 
```
First, rewinding head to replay your work on top of it...
could not detach HEAD

**Comment by sem-geologist** [comment]
ok, I think I did it...
I had already new branch ready with copy of changes in case it would not work...
git sometimes makes me wanting to detach my real head.
5 checks? is this normal (double appveyor and travis...)

**Comment by sem-geologist** [comment]
oh... yeah.. it actually went completely wrong... ok, I am closing this, and reopening with other branch...

**Pull request closed without merging**